### DISCUSSION TITLE
Building SignServer(Community) with Maven

### DISCUSSION THREAD

**Discussion by KolRe01** [Keyfactor/signserver-ce#73] [General]
Hi,
anyone tried to build signserver(community) with maven? The documentation (signserver/doc/htdocs/Building_SignServer_with_Maven.html) describes how to build the signserver but the x509-common-util dependency is not available in the central maven repo. An mvn dependency:resolve gets:
...The POM for com.keyfactor:x509-common-util:jar:0.7.4 is missing, no dependency information available...

I tried to build version 6.2.0.Beta1-SNAPSHOT. The final version must have the same problem.

It's possible that i overlooked a hint? Don't woud like to tinker the pom using the missing lib/dependency from the binary distribution. 

Thanks for any suggestions :)
René

**Comment by mlundblad** [comment]
Hi there!

There is a script ./lib/maven-install-files.sh

Running this will install the custom dependencies in your local Maven repo. These JARs are bundled inside the lib directory (and they are packaged in the .ear file when running ant deploy directly from the bundled files, they just needs to be "installed" for Maven to be able to build the sources.

//Marcus

**Reply by KolRe01** [reply]
Hi Marcus, thanks for the tip. Using the script, I got a step further. But I get still an error:
"... \cesecore-common-7.0.0.1signserver6.2.0.jar' does not exist ..."
The script: maven-install-files.sh tries to install cesecore-common-7.0.0.1signserver6.2.0.jar from the .../lib/ext directory instaed of  ../lib/ext/cesecore-ce  

Finally the servicemanifestbuilder:jar:1.0.2 is not available with the packed source code: https://github.com/Keyfactor/signserver-ce/archive/refs/tags/v6.2.0.Final.zip

Perhaps I overlooked something again?
Regards René

**Reply by mlundblad** [reply]
Btw, did you run "ant init" before building?

You could try something like:

```
$ rm res/edition.properties
$ ant clean init
```

And then try running Maven again

Init will detect the edition (in this case CE, or when using the commercial version EE) and copy in the appropriate dependencies that are edition-specific.

(needs to remove the properties file, otherwise it will not do the copying again, I think looking at the paths you're on Windows, so you might run some command like "del", but you get the idea…)

**Reply by KolRe01** [reply]
thanks, it works :)
